### DISCUSSION TITLE
Include puml file in pages

### DISCUSSION THREAD

**Discussion by TCke83** [bsorrentino/maven-confluence-plugin#276] [Q&A] [answered]
Hi,

i'm struggeling to include a puml file in a page through the maven plugin on a markdown page.

I've added it as an attachment to my page and that seems to be correct as i can see the attachment.

But the visualisation on the page does not work. I use following snippet to include the puml file in my markdown page:

`![Domain model](domain.urm.puml "Domain model")`

Attachement is added to page with (in site.xml):

`<attachment name="domain.urm.puml" uri="pages/images/domain.urm.puml" version="1"/>`

Do i have to do this in another manner? Or should this work?

Thanks for your reply.

Kind regards,

Tom Cassimon

**Comment by bsorrentino** [comment, marked as answer]
hi @TCke83 

currently the plugin doesn't include build-in support for PlantUML. 

The only way that you have is to convert `puml` in png` before publication.

Long time ago I wrote this article [PlantUML & Maven - Enrich your project's documentation 
](http://soulsoftware-bsc.blogspot.com/2014/01/plantuml-maven-enrich-your-projects.html) that hopefully could help you
